### ISSUE TITLE
Handle ~~"predictive back"~~ ActivityTransitionAnimation deprecation in API34

### ISSUE THREAD

**Issue by BrayanDSO** [ankidroid/Anki-Android#14558]
> [!IMPORTANT]
> This issue is to be included in a research study, and should not be worked on

API34 adds new predictive back behavior, which implies that the transition animation overrides for activity stack change animations now need to know whether the animation override is for activity open or close, whereas we override both as a unit without thought

This needs a solution before the deprecated methods go away, likely in the form of plumbing through (via a new parameter) the idea of whether the call site intended to override open, close, or both animations. For backwards compatibility this will likely need to be a Compat interface (or interfaces) to replace the current implementation

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android/pull/14556#discussion_r1364530599_

**Comment by BrayanDSO** [comment]
Predictive back gesture reference: https://developer.android.com/guide/navigation/custom-back/predictive-back-gesture

**Comment by lukstbit** [comment]
We also need to fix the onBackPressed() deprecation in #12364 .

**Comment by mikehardy** [comment]
@lukstbit yep  - just had to handle that to pass lint in the stack of PRs, commit here - and examination of that commit (probably altering it or the objects slightly...) should be part of the success criteria to close this issue https://github.com/ankidroid/Anki-Android/pull/14518/commits/594290e7f3676083d06b9e0500a53ca2ce8898b0

**Comment by mikehardy** [comment]
PR #13586 looks like the proper way to go about things - sad (word directed at myself) that the PR has sat so long and accumulated some conflicts! It looks good and should be considered for fixup + merge to fix this for real

**Comment by mikehardy** [comment]
Just adding a quick note here that one item has been provisionally handled, so it may not show up in the deprecation scans anymore, but it needs a more thorough handling as it's a temporary patch:

https://github.com/ankidroid/Anki-Android/pull/14712/files#diff-6d9f7ea7bba9cc72513b5078633cee9bdbbd1ea7646114869679f302ad607459R277-R279

**Comment by david-allison** [comment]
* Opted in to predictive back in: https://github.com/ankidroid/Anki-Android/pull/17826

Reviewer and DeckPicker both have 'exit' callbacks, which take priority over closing the activity, so the animations aren't working as expected

**Comment by david-allison** [comment]
This issue was selected for research and an implementation is pending.

I will take the PR over the finish line if necessary in ~2 weeks
